pkg/file/groups.go

```diff
@@ -60,6 +60,26 @@ func (gs *Groups) matchExistingGroup(format *CompiledFormat, fileMatch bool, loc
 	return false
 }
 
+func (gs *Groups) FilterGroupsByStrictness(strictness int) {
```

Comment:
I think the heaviest part of the testing of this logic should be done in `groups_test.go`. since of where it lives. In `groups.go`